packages/files-ui/src/Contexts/FilesContext.tsx

```diff
@@ -583,6 +621,104 @@ const FilesProvider = ({ children }: FilesContextProps) => {
         .catch(console.error)
     }, [filesApiClient, encryptForPublicKey, publicKey, refreshBuckets])
 
+  const transferFileBetweenBuckets = useCallback(async (
+    sourceBucketId: string,
+    sourceFile: FileSystemItem,
+    path: string,
+    destinationBucket: BucketKeyPermission,
+    keepOriginal = false
+  ) => {
+    const toastId = uuidv4()
+    const UPLOAD_PATH = "/"
+
+    const transferProgress: TransferProgress = {
+      id: toastId,
+      complete: false,
+      error: false,
+      progress: 0,
+      operation: "Download"
+    }
+    dispatchTransfersInProgress({ type: "add", payload: transferProgress })
+
+    getFileContent(sourceBucketId, {
+      cid: sourceFile.cid,
+      file: sourceFile,
+      path: getPathWithFile(path, sourceFile.name),
+      onDownloadProgress: (progressEvent) => {
+        dispatchTransfersInProgress({
+          type: "progress",
+          payload: {
+            id: toastId,
+            progress: Math.ceil(
+              (progressEvent.loaded / sourceFile.size) * 50
+            )
+          }
+        })
+      }
+    }).then(async (fileContent) => {
+      if (!fileContent) {
+        dispatchTransfersInProgress({
+          type: "error",
+          payload: {
+            id: toastId,
+            errorMessage: "An error occured while downloading the file"
+          }
+        })
+        return
+      }
+
+      dispatchTransfersInProgress({
+        type: "operation",
+        payload: {
+          id: toastId,
+          operation: "Encrypt & Upload"
+        }
+      })
+
+      await encryptAndUploadFiles(
+        destinationBucket,
+        [new File([fileContent], sourceFile.name, { type: sourceFile.content_type })],
+        UPLOAD_PATH,
+        (progressEvent) => {
+          dispatchTransfersInProgress({
+            type: "progress",
+            payload: {
+              id: toastId,
+              progress: Math.ceil(
+                50 + (progressEvent.loaded / sourceFile.size) * 50
+              )
+            }
+          })
+        }
+      )
+
+      if (!keepOriginal) {
+        await filesApiClient.removeBucketObject(sourceBucketId, { paths: [getPathWithFile(path, sourceFile.name)] })
+      }
+
+      dispatchTransfersInProgress({
+        type:"complete",
+        payload: {
+          id: toastId
+        }
+      })
+      return Promise.resolve()
+    }).catch((error) => {
+      dispatchTransfersInProgress({
+        type: "error",
+        payload: {
+          id: toastId,
+          errorMessage: `An error occured: ${error}`
```

Comment:
translation and typo
```suggestion
          errorMessage: t`An error occurred: ${error}`
```